test/example_test.exs

```diff
@@ -0,0 +1,77 @@
+defmodule Example.Test do
+  @moduledoc false
+  use ExUnit.Case
+
+  setup do
+    token = "test_token"
+    Application.put_env(:manifest, :auth_token, token)
+    on_exit(fn -> Application.delete_env(:manifest, :auth_token) end)
+    {:ok, pid} = Cache.start_link()
+    [pid: pid, token: token]
+  end
+
+  test "example of success", %{pid: pid, token: token} do
+    id = 5
+
+    results =
+      Manifest.new()
+      |> Manifest.add_step(:cache_read, fn _ ->
+        {:ok, :no_rollback, Cache.lookup(pid, token, id)}
+      end)
+      |> Manifest.add_step(:database_read, fn _ -> {:ok, :no_rollback, mock_lookup(id)} end)
+      |> Manifest.add_step(
+        :cache_put,
+        fn %{database_read: %{id: id} = record} ->
+          Cache.put(pid, token, id, record)
+          {:ok, id}
+        end,
+        fn id ->
+          Cache.delete(pid, token, id)
+          {:ok, id}
+        end
+      )
+      |> Manifest.perform()
```

Comment:
end result is pretty cool structure-wise, seems to behave/is used in a similar way to a Multi call (with the added benefit of no db lockdowns). I can appreciate the easy-transition 🙂